libbeat/docs/shared-env-vars.asciidoc

```diff
@@ -49,3 +67,36 @@ and what each configuration looks like after replacement:
 |`name: ${NAME:beats}` |no setting            |`name: beats`
 |`name: ${NAME:beats}` |`export NAME=elastic` |`name: elastic`
 |==================================
+
+[float]
+=== Specifying Complex Objects in Environment Variables
+
+You can specify complex objects, such as lists or dictionaries, in environment
+variables by using a JSON-like syntax.
+
+For example, the following environment variable is set to a list:
+
+[source,yaml]
+-------------------------------------------------------------------------------
+ES_HOSTS="10.45.3.2:9220,10.45.3.1:9230"
+-------------------------------------------------------------------------------
+
+You can reference this variable in the config file:
+
+[source,yaml]
+-------------------------------------------------------------------------------
+output.elasticsearch:
+  hosts: '${ES_HOSTS}'
+-------------------------------------------------------------------------------
+
+When the Beat loads the config file, it resolves the environment variable and
+replaces it with the specified list before reading the `hosts` setting.
+
+When you specify complex objects in environment variables, keep the following
+in mind:
+
+* Strings can be unquoted, single-quoted, or double-quoted
+* Arrays at the top-level do not require brackets (`[]`). Instead, use commas (`,`) to
+separate the elements.
```

Comment:
Sounds to me like we discourage (don't allow/want) users to use `[]` for top-level arrays. Especially for more complex objects I'd encourage `[]` even on top-level. It's just `[]` is optional for top-level arrays for user convience only.